chat/consumers.py

```diff
@@ -30,16 +34,63 @@ class InvestorStartupMessageConsumer(AsyncWebsocketConsumer):
       - Sending/receiving messages via WebSocket in real-time.
 
     Attributes:
-        room_name (str): Identifier for the messaging room, extracted from the WebSocket URL.
-        room_group_name (str): Group name used by Channels to broadcast messages.
+        user (Optional[User]): Current user connected to the WebSocket.
+        other_user (Optional[User]): The other participant in the chat.
+        room (Optional[Room]): The database Room object for this chat.
+        room_name (Optional[str]): Identifier for the messaging room, extracted from the WebSocket URL.
+        room_group_name (Optional[str]): Group name used by Channels to broadcast messages.
     """
-
-    async def connect(self):
+    user: Optional[User]
+    other_user: Optional[User]
+    room: Optional[Room]
+    room_name: Optional[str]
+    room_group_name: Optional[str]
+
+    def __init__(self, *args, **kwargs):
+        """ Initialize the WebSocket consumer with default attributes. """
+        super().__init__(*args, **kwargs)
+        self.user: Optional[User] = None
+        self.other_user: Optional[User] = None
+        self.room: Optional[Room] = None
+        self.room_name: Optional[str] = None
+        self.room_group_name: Optional[str] = None
+
+    async def connect(self) -> None:
         """
-        Handles a new WebSocket connection.
-        Joins the corresponding room group based on a sanitized room name.
-        It allows only alphanumeric, dash, underscore.
+        Handles a new WebSocket connection between two users.
+
+        Workflow:
+        1. Validates that the current user (`self.user`) is authenticated.
+        2. Retrieves the `other_user` from the URL and validates their existence.
+        3. Creates or retrieves a chat room object (MongoEngine-backed).
+        4. Sanitizes the `room_name` from the URL to allow only alphanumeric,
+           dash (-), and underscore (_), and truncates to 50 characters max.
+        5. Adds the WebSocket channel to the appropriate group.
+        6. Accepts the connection if all checks pass.
+
+        Custom WebSocket close codes used:
+        - 4401 → Unauthorized: the connecting user is not authenticated.
+        - 4404 → Not Found: the other user with the provided ID does not exist.
+        - 4000 → Invalid room name: the `room_name` contains only invalid characters.
+        - 1011 → Internal Error: failed to add channel to group (unexpected error).
         """
+        self.user = self.scope['user']
```

Comment:
Raw user extraction does not check for key existence, risking KeyError. Add a default fallback for missing 'user' key.
```
self.user = self.scope.get('user', None)
```